HiP-FeatureToggle/Controllers/FeatureGroupsController.cs

```diff
@@ -0,0 +1,125 @@
+    /// </summary>
+    [Authorize]
+    [Route("api/[controller]")]
+    public class FeatureGroupsController : Controller
+    {
+        private readonly FeatureGroupsManager _manager;
+        private readonly CmsService _cmsService;
+
+        public FeatureGroupsController(FeatureGroupsManager manager, CmsService cmsService)
+        {
+            _manager = manager;
+            _cmsService = cmsService;
+        }
+
+        [HttpGet]
+        public IActionResult GetAll([FromQuery]string identity)
+        {
+            if (!IsAdministrator(identity))
+                return Forbid();
+
+            var groups = _manager.GetGroups(loadMembers: true, loadFeatures: true);
+            var results = groups.ToList().Select(g => new FeatureGroupResult(g)); // note: ToList() is required here
+            return Ok(results);
+        }
+
+        [HttpGet("{groupId}")]
+        public IActionResult GetById([FromQuery]string identity, int groupId)
+        {
+            if (!IsAdministrator(identity))
+                return Forbid();
+
+            var group = _manager.GetGroup(groupId, loadMembers: true, loadFeatures: true);
+
+            if (group == null)
+                return NotFound();
+
+            return Ok(new FeatureGroupResult(group));
+        }
+
+        [HttpPost]
+        public IActionResult Create([FromQuery]string identity, [FromBody]FeatureGroupViewModel groupVM)
+        {
+            if (!IsAdministrator(identity))
+                return Forbid();
+
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            try
+            {
+                var newGroup = new FeatureGroup { Name = groupVM.Name };
+
+                newGroup.EnabledFeatures = _manager.GetFeatures(groupVM.EnabledFeatures)
+                    .Select(f => new FeatureToFeatureGroupMapping(f, newGroup))
+                    .ToList();
+
+                _manager.AddGroup(newGroup);
+                return Ok();
+            }
+            catch (ArgumentException e)
+            {
+                return BadRequest(e.Message);
+            }
+        }
+
+        [HttpDelete("{groupId}")]
+        public IActionResult Delete([FromQuery]string identity, int groupId)
+        {
+            if (!IsAdministrator(identity))
+                return Forbid();
+
+            var success = _manager.RemoveGroup(groupId);
+
+            if (!success)
+                return NotFound();
+
+            return Ok();
+        }
+
+        [HttpPut("{groupId}")]
+        public IActionResult Update([FromQuery]string identity, int groupId, [FromBody]FeatureGroupViewModel groupVM)
+        {
+            if (!IsAdministrator(identity))
+                return Forbid();
+
+            // TODO: What is the purpose of this operation?
+            return BadRequest();
+        }
+
+        [HttpPut("/api/Users/{userId}/FeatureGroup/{groupId}")]
```

Comment:
You did not design the API so that's not your fault, but I think the URL does not describe at all that the user is moved to the given group if I call this... we can discuss it later in person